File: transform.js
const fs = require('fs');

const data = JSON.parse(fs.readFileSync('src/data/questions/XII_Securities Markets Foundation.json', 'utf8'));

let new_data = [];

let id_counter = 20001;

for (let q of data) {

    let question = q.question;

    let options = q.options;

    let correct = q.correct_answer;

    let new_options = [];

    let correct_id = null;

    for (let i = 0; i < options.length; i++) {

        let opt_id = String.fromCharCode(97 + i);

        new_options.push({ id: opt_id, text: options[i] });

        if (options[i] === correct || (correct.startsWith("Option ") && correct[7].toLowerCase() === opt_id)) {

            correct_id = opt_id;

        }

    }

    new_data.push({

        id: id_counter,

        question: question,

        options: new_options,

        correct_answer: correct_id

    });

    id_counter++;

}

fs.writeFileSync('src/data/questions/XII_Securities Markets Foundation.json', JSON.stringify(new_data, null, 4));

File: transform.py
import json

with open('src/data/questions/II-A.json', 'r') as f:
    data = json.load(f)

new_data = []
id_counter = 20001

for q in data:
    question = q['question']
    options = q['options']
    correct = q['correct_answer']
    
    # Map options to a, b, c, d
    new_options = []
    correct_id = None
    for i, opt in enumerate(options):
        opt_id = chr(ord('a') + i)
        new_options.append({"id": opt_id, "text": opt})
        if opt == correct or (correct.startswith("Option ") and correct[7].lower() == opt_id):
            correct_id = opt_id
    
    new_q = {
        "id": id_counter,
        "question": question,
        "options": new_options,
        "correct_answer": correct_id
    }
    new_data.append(new_q)
    id_counter += 1

with open('src/data/questions/II-A.json', 'w') as f:
    json.dump(new_data, f, indent=4)
